### ISSUE TITLE
Support version ranges natively

### ISSUE THREAD

**Issue by reubano** [pyupio/pyup#203]
Taking the example from #50 further, if your package *does* depend on `requests`, and you don't want a downstream lib preventing you from using a new version.

`your-package-requirements.txt`
```
some-package=1.0.3
requests >=2.7  # some-package uses requests=2.5, but I want a newer version
```

This may cause a version conflict in pip unless `some-package` has `requests>=2.5,<3.0` in `requirements.txt`. But now that `requests` isn't "pinned" in `some-package`, the [security checks don't work](https://pyup.io/account/repos/github/nerevu/riko/). 

I read that you [support filters](https://pyup.io/docs/filter/), so `some-package` could do

```
requests  # pyup: >=2.5,<3.0
```

But that would require rewriting all ranged requirements. Would there be a downside to natively recognizing?

```
requests >=2.5,<3.0
```

**Comment by jayfk** [comment]
> But that would require rewriting all ranged requirements. Would there be a downside to natively recognizing?

Yes. You would no longer receive update pull requests for the range. Requests is a good example here: Every once in a while they issue a security release. You would no longer know about that from the changelogs the bot adds to the PR.

Another downside is that you are no longer able to tell what releases are installed on your servers by looking at the requirements file alone. This entirely depends on what the latest release was during your last pip install.

Maybe we could make this configurable in some way?
